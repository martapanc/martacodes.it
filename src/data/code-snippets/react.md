```
<MartaPancaldi
  role={"Software Engineer"}
  skills={["NodeJS", "Angular", "React"]}
  favouriteColor={"#0047ab"}
/>
```
