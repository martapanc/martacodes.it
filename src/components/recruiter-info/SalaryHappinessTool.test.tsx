import { fireEvent, render, waitFor } from '@testing-library/react';
import React from 'react';

import {
  SalaryHappinessProps,
  SalaryHappinessTool,
} from '@/components/recruiter-info/SalaryHappinessTool';

describe('Salary', () => {
  const testRange = {
    min: 40000,
    median: 80000,
    max: 120000,
  };

  const config = {
    displayTitle: true,
    currency: 'EUR',
    forexMultiplier: 1.2,
  };

  const props: SalaryHappinessProps = {
    salaryData: testRange,
    config,
  };

  it('renders correctly', () => {
    const { container } = render(<SalaryHappinessTool {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('switches the emoji when changing the range', async () => {
    const { getByTestId } = render(<SalaryHappinessTool {...props} />);
    const rangeInput = getByTestId('salaryValue');
    const emoji = getByTestId('emoji');
    expect(emoji).toHaveAccessibleDescription('getting there');

    fireEvent.change(rangeInput, { target: { value: '40000' } });
    await waitFor(() => {
      expect(emoji).toHaveAccessibleDescription(/are you serious?/i);
    });

    fireEvent.change(rangeInput, { target: { value: '50000' } });
    expect(emoji).toHaveAccessibleDescription(/way too low/i);

    fireEvent.change(rangeInput, { target: { value: '60000' } });
    expect(emoji).toHaveAccessibleDescription(/too low/i);

    fireEvent.change(rangeInput, { target: { value: '70000' } });
    expect(emoji).toHaveAccessibleDescription(/getting there/i);

    fireEvent.change(rangeInput, { target: { value: '80000' } });
    expect(emoji).toHaveAccessibleDescription(/pretty good/i);

    fireEvent.change(rangeInput, { target: { value: '90000' } });
    expect(emoji).toHaveAccessibleDescription(/even better/i);

    fireEvent.change(rangeInput, { target: { value: '100000' } });
    expect(emoji).toHaveAccessibleDescription(/that's amazing/i);

    fireEvent.change(rangeInput, { target: { value: '110000' } });
    expect(emoji).toHaveAccessibleDescription(/make it rain!/i);
  });
});
