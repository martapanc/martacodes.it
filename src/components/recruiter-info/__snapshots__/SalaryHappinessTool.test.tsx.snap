// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salary renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1v5z18m"
  >
    <h3>
      Salary Happiness Tool
    </h3>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
      data-testid="salaryValue"
    >
      <span
        class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
      />
      <span
        class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
        style="left: 0%; width: 31.25%;"
      />
      <span
        class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
        data-index="0"
        style="left: 31.25%;"
      >
        <input
          aria-label="Default"
          aria-orientation="horizontal"
          aria-valuemax="120000"
          aria-valuemin="40000"
          aria-valuenow="65000"
          data-index="0"
          max="120000"
          min="40000"
          step="5000"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="65000"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
        >
          <span
            class="MuiSlider-valueLabelCircle"
          >
            <span
              class="MuiSlider-valueLabelLabel"
            >
              65000
            </span>
          </span>
        </span>
      </span>
    </span>
    <label
      for="salaryValue"
    >
      ‚Ç¨65,000.00
    </label>
    <p>
      <strong>
        Happiness Score:
      </strong>
      <span
        class="text-3xl"
        data-testid="emoji"
        title="Getting there"
      >
         üòè
      </span>
    </p>
  </div>
</div>
`;
